src/filters/filter_chain.h

```diff
@@ -54,8 +56,9 @@ class FilterChainImpl : public FilterChain {
  private:
   unsigned int threads_;
   config::FilterChain config_;
-  std::shared_ptr<oidc::SessionStore> oidc_session_store_;
-  std::vector<std::shared_ptr<oidc::JwksResolver>> jwks_resolver_map_;
+  oidc::SessionStorePtr oidc_session_store_;
+  oidc::JwksResolverCachePtr jwks_resolver_cache_;
+  std::vector<FilterFactoryPtr> filter_factory_chain_;
```

Comment:
to its semantics, this would be better named as `filter_factories_` right? Chain has its own meaning and here can have audience think it's related to the filter chain. Simple plural form should be enough.